daemon/libnetwork/drivers/bridge/port_mapping_linux_test.go

```diff
@@ -1050,3 +992,33 @@ func (c *mockPortDriverClient) AddPort(_ context.Context, proto string, hostIP,
 		return nil
 	}, nil
 }
+
+type stubPortMapper struct {
+	reqs   [][]portmapperapi.PortBindingReq
+	mapped []portmapperapi.PortBinding
+}
+
+func (pm *stubPortMapper) MapPorts(_ context.Context, reqs []portmapperapi.PortBindingReq, _ portmapperapi.Firewaller) ([]portmapperapi.PortBinding, error) {
+	if len(reqs) == 0 {
+		return []portmapperapi.PortBinding{}, nil
+	}
+	pm.reqs = append(pm.reqs, reqs)
+	pbs := sliceutil.Map(reqs, func(req portmapperapi.PortBindingReq) portmapperapi.PortBinding {
+		return portmapperapi.PortBinding{PortBinding: req.PortBinding}
+	})
+	pm.mapped = append(pm.mapped, pbs...)
+	return pbs, nil
+}
+
+func (pm *stubPortMapper) UnmapPorts(_ context.Context, reqs []portmapperapi.PortBinding, _ portmapperapi.Firewaller) error {
+	for _, req := range reqs {
+		idx := slices.IndexFunc(pm.mapped, func(pb portmapperapi.PortBinding) bool {
+			return pb.Equal(&req.PortBinding)
+		})
+		if idx == -1 {
+			return fmt.Errorf("stubPortMapper.UnmapPorts: pb doesn't exist %v", req)
+		}
+		pm.mapped = slices.Delete(pm.mapped, idx, idx)
```

Comment:
Stumbled on this in another PR;

```suggestion
		pm.mapped = slices.Delete(pm.mapped, idx, idx+1)
```


In case it's causing issues and this PR ever gets backported; opened a PR with a fix;

- https://github.com/moby/moby/pull/50703